ddtrace/sampler.py

```diff
@@ -199,6 +200,34 @@ def __init__(
             )
             self.default_sampler = SamplingRule(sample_rate=default_sample_rate)
 
+    def parse_from_env_variable(self):
+        rules = []
+        # DD_TRACE_SAMPLING_RULES = ['{"sample_rate": "1", "service": "b", "name": 0.1}', ....]
+        env_rules = get_env("trace", "sampling_rules")
+        if env_rules is not None:
+            json_rules = []
+            try:
+                json_rules = json.loads(env_rules)
+            except json.decoder.JSONDecodeError:
+                log.debug(
+                    "Unable to set DD_TRACE_SAMPLING_RULES={%s} due to parsing configuration", env_rules, exc_info=True
+                )
+            # setting each rule from JSOB object
+            for rule in json_rules:
+                try:
+                    sample_rate = float(rule["sample_rate"])
+                    service = rule.get("service", SamplingRule.NO_RULE)
+                    name = rule.get("name", SamplingRule.NO_RULE)
+                    sampling_rule = SamplingRule(sample_rate=sample_rate, service=service, name=name)
+                except ValueError:
+                    log.debug("Invalid value while setting SamplingRule", exc_info=True)
+                    continue
+                except KeyError:
+                    log.debug("No Sampling rate provided", exc_info=True)
+                    continue
```

Comment:
instead we might want to do:

```python
if "sample_rate" not in rule:
    raise / log / continue ?
```

the reason for this is. if we ended up with a `KeyError` from something else we would still print "No sampling rate provided" which might be accurate.

Reply:
Thats a good point on the KeyError exception, but how to handle should be same as your other two suggestions. Is it better to have an application raise an exception for the user to handle immediately or to log and error and move on since this particular rule since it is configured incorrectly?

Reply:
In this case we wait to fail fast and hard (raise an exception for the user to deal with), because otherwise we'll result in invalid configuration

Reply:
👍